mobile/lib/pages/editing/edit.page.dart

```diff
@@ -0,0 +1,110 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:fluttertoast/fluttertoast.dart';
+import 'package:immich_mobile/pages/editing/crop.page.dart';
+import 'package:immich_mobile/entities/asset.entity.dart';
+import 'package:immich_mobile/widgets/common/immich_image.dart';
+import 'package:immich_mobile/widgets/common/immich_toast.dart';
+import 'package:immich_mobile/providers/api.provider.dart';
+import 'package:immich_mobile/services/api.service.dart';
+
+@immutable
+class EditImagePage extends StatelessWidget {
+  final dynamic imageSource; // Can be either Asset or Image
+  const EditImagePage({super.key, required this.imageSource});
+
+  void _navigateToCropImagePage(BuildContext context) {
+    Navigator.of(context).push(
+      MaterialPageRoute(
+        builder: (context) => CropImagePage(image: _getImageWidget()),
+      ),
+    );
+  }
+
+  ImageProvider _getImageProvider() {
+    if (imageSource is Asset) {
+      return ImmichImage.imageProvider(asset: imageSource);
+    } else if (imageSource is Image) {
+      return (imageSource as Image).image;
+    } else {
+      throw Exception('Invalid image source type');
+    }
+  }
+
+  Image _getImageWidget() {
+    if (imageSource is Asset) {
+      return Image(image: ImmichImage.imageProvider(asset: imageSource));
+    } else if (imageSource is Image) {
+      return imageSource;
+    } else {
+      throw Exception('Invalid image source type');
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final ImageProvider provider = _getImageProvider();
+
+    return Scaffold(
+      appBar: AppBar(
+        backgroundColor: Colors.black,
```

Comment:
Do we want to be brightness aware here or prefer to use a solid black backdrop?

Reply:
I dont have a preference, but I think solid black looks good. I will let you decide this one.